src/field/pokemon.ts

```diff
@@ -4068,6 +4077,46 @@ export abstract class Pokemon extends Phaser.GameObjects.Container {
     return true;
   }
 
+  protected setStatus(
+    effect: StatusEffect,
+    { toxicTurnCount = 0, sleepTurnsRemaining = 0 }: { toxicTurnCount?: number; sleepTurnsRemaining?: number },
+  ): void {
+    this.status = {
+      effect,
+      toxicTurnCount,
+      sleepTurnsRemaining,
+    };
+  }
+
+  public advanceStatusCounter(): void {
```

Comment:
We should stick to already used terms in the project, which would be `lapse` in this case

```suggestion
  public lapseStatusCounter(): void {
```

Reply:
Also - NIT: Missing tsdocs

Reply:
Oops, forgot about that.

Reply:
This doesn't do the same thing though, because it also ticks up the toxic counter which is different from `[Battler|Arena]Tag`s.

Reply:
I see. Keep it then